Release Candidate 1

cholmes opened the pull request:
Updating the repo to be ready for 1.0.0-rc.1

I believe this just needs to be the readme, though we may need some changes to publish the new parquet compatibility document on the website (I don't think we need to publish right when we publish rc.1, it can be a follow up imho).

We also need to wait on https://github.com/opengeospatial/geoparquet/pull/175 as that should be 1.0.0-rc.1 if it lands.

m-mohr commented:
What are the actual changes to the spec between beta and rc? The PR looks like it's just additional documentation, but no spec change. But in that case I'm wondering why releasing a new version?

cholmes commented:
The core geoparquet spec doesn't have any changes, but I think we still should put out an rc.1 to say 'this is your last chance to give feedback on it'.

The bit that has been done since beta 1 is the 'compatibility' document, at https://github.com/opengeospatial/geoparquet/pull/172

For this repo we don't have a dev -> master merge for 'release'. Release is to tag it and then the spec gets published to the website at the appropriate version. So the readme update is just to get it ready for that.

m-mohr commented:
Thanks.

It just feels a bit weird to update a spec version that has not changed, which may need changes in code or files to be on the latest version number although actually it's not needed.

The "last chance" could also be blog posts/tweets/OGC communication/..., right?

But just my 2cts.

cholmes commented:
Let's use approval of this request to mean approval to cut the 1.0.0-rc.1 release. So once we merge this I'll tag it and cut the rc.1 release.

cholmes commented:
Draft blog post is up at https://medium.com/@cholmes/geoparquet-1-0-0-rc-1-released-26df7188bec5 - any feedback is appreciated (and I'll give it one more review before I publish).

kylebarron commented:
> 



> Draft blog post is up at https://medium.com/@cholmes/geoparquet-1-0-0-rc-1-released-26df7188bec5 - any feedback is appreciated (and I'll give it one more review before I publish).

It is annoying that Medium has a hard loginwall now. I wasn't able to view this without signing in to Medium (or maybe because it's a draft?).

rouault commented:
> It may require changes in code or files to be on the latest version number although actually there's no actual change.

will the version number in schema.json change upon rc1 release ? I agree it is slightly annoying to cause code and data files to change without actual schema change, but I can live with that...

cholmes commented:
>> Draft blog post is up at https://medium.com/@cholmes/geoparquet-1-0-0-rc-1-released-26df7188bec5 - any feedback is appreciated (and I'll give it one more review before I publish).

> It is annoying that Medium has a hard loginwall now. I wasn't able to view this without signing in to Medium (or maybe because it's a draft?).

Hrm, I hope it's just because it's a draft, which I guess makes some sense? I've shared drafts before, but maybe everyone just had log-ins? Or they changed something. If the actual post requires login I'll put it somewhere else.

kylebarron commented on README.md:
```diff
@@ -2,19 +2,19 @@
+This repository defines a [specification](https://geoparquet.org/releases/) for how to store geospatial [vector data](https://gisgeography.com/spatial-data-types-vector-raster/) (point, lines, polygons) in [Apache Parquet](https://parquet.apache.org/), a popular columnar storage format for tabular data - see [this vendor explanation](https://databricks.com/glossary/what-is-parquet) for more on wh ...
 
-**Warning:** This specification is currently in 1.0 'beta' releases, which means we don't anticipate anything major will change, but still reserve the right to make changes in backwards incompatible ways. See the [versioning](#versioning) section below for more info. If you are excited about the potential please collaborate with us by building implementations, sounding in on the issues and contri ...
+**Note:** This specification is currently in 1.0 'release candidate' status, which means the community is proposing the current version to be 1.0.0, and if no blocking negative feedback is made in the next few weeks then it will become 1.0.0. This means breaking changes are still possible, but quite unlikely - see the [versioning](#versioning) section below for more info.
```
> in the next few weeks

I find "in the next few weeks (as of late July)" to be helpful so that if someone sees this repo in two weeks, they know they're almost at the end of the time period.

cholmes replied:
```suggestion
**Note:** This specification is currently in 1.0 'release candidate' status, which means the community is proposing the current version to be 1.0.0, and if no blocking negative feedback is made in the next few weeks (as of late July) then it will become 1.0.0. This means breaking changes are still possible, but quite unlikely - see the [versioning](#versioning) section below for more info.
```

kylebarron commented on README.md:
```diff
@@ -2,19 +2,19 @@
-Initial work started in the [geo-arrow-spec](https://github.com/geopandas/geo-arrow-spec/) GeoPandas repository, and that will continue on
-Arrow work in a compatible way, with this specification focused solely on Parquet.
+Anyone is welcome to join the project, by building implementations, trying it out, giving feedback through issues and contributing to the spec via pull requests.
+Initial work started in the [geo-arrow-spec](https://github.com/geoarrow/geoarrow) GeoPandas repository, and that will continue on
```
```suggestion
Initial work started in the [geoarrow](https://github.com/geoarrow/geoarrow) repository, and that will continue on
```

kylebarron commented on README.md:
```diff
@@ -54,7 +53,10 @@ will work much better if it is backing a system that is constantly updating the
 ## Roadmap
 
-Our aim is to get to a 1.0.0 in the next few months. The next steps are to push towards our 1.0.0 acceptance criteria, articulated in this [discussion post](https://github.com/opengeospatial/geoparquet/discussions/122), which mostly involves getting great validation tools, more software implementations, and more data providers. This should provide key feedback to feel confident in going to 1.0. T ...
+Our aim is to get to a 1.0.0 final in the next few weeks. The goal of 1.0.0 is to establish a baseline of interoperability for geospatial information in Parquet. For 1.0.0
```
```suggestion
Our aim is to get to a 1.0.0 final by the end of August 2023. The goal of 1.0.0 is to establish a baseline of interoperability for geospatial information in Parquet. For 1.0.0
```

kylebarron commented on README.md:
```diff
@@ -54,7 +53,10 @@ will work much better if it is backing a system that is constantly updating the
+Our aim is to get to a 1.0.0 final in the next few weeks. The goal of 1.0.0 is to establish a baseline of interoperability for geospatial information in Parquet. For 1.0.0
+the only geometry encoding option is Well Known Binary, but we made it an option to allow other encodings. The main goal of 1.1.0 will be to incorporate a more columnar-oriented
+geometry format, which is currently being worked on as part of the [GeoArrow spec](https://github.com/geoarrow/geoarrow). Once that gets finalized we will add the option to
+GeoParquet. In general 1.1.0 will further explore spatial optimization, spatial indices and spatial partitioning to improve GeoParquet's.
```
```suggestion
GeoParquet. In general 1.1.0 will further explore spatial optimization, spatial indices and spatial partitioning to improve GeoParquet's performance.
```

kylebarron reviewed (commented):
How do we want to handle the open issues that [currently have 1.0.0-beta.2](https://github.com/opengeospatial/geoparquet/milestone/8) as a milestone? Should those be closed before this PR is merged? Or updated to have 1.0.0 as the milestone?

cholmes commented:
> How do we want to handle the open issues that [currently have 1.0.0-beta.2](https://github.com/opengeospatial/geoparquet/milestone/8) as a milestone? Should those be closed before this PR is merged? Or updated to have 1.0.0 as the milestone?

Ah, good point. Just went through that milestone and renamed it 1.0.0-rc.1, and moved issues off to 1.0.1 I think we could do some between rc.1 and 1.0.0, since they don't affect the spec itself, but I'm not sure we'll get to them.

m-mohr commented on README.md:
```diff
@@ -2,19 +2,19 @@
+This repository defines a [specification](https://geoparquet.org/releases/) for how to store geospatial [vector data](https://gisgeography.com/spatial-data-types-vector-raster/) (point, lines, polygons) in [Apache Parquet](https://parquet.apache.org/), a popular columnar storage format for tabular data - see [this vendor explanation](https://databricks.com/glossary/what-is-parquet) for more on wh ...
 
-**Warning:** This specification is currently in 1.0 'beta' releases, which means we don't anticipate anything major will change, but still reserve the right to make changes in backwards incompatible ways. See the [versioning](#versioning) section below for more info. If you are excited about the potential please collaborate with us by building implementations, sounding in on the issues and contri ...
+**Note:** This specification is currently in 1.0 'release candidate' status, which means the community is proposing the current version to be 1.0.0, and if no blocking negative feedback is made in the next few weeks (as of late July) then it will become 1.0.0. This means breaking changes are still possible, but quite unlikely - see the [versioning](#versioning) section below for more info.
```
```suggestion
**Note:** This specification is currently in 1.0 'release candidate' status, which means the community is proposing the current version to be 1.0.0, and if no blocking negative feedback is made until end of August 2023 then it will become 1.0.0. This means breaking changes are still possible, but quite unlikely - see the [versioning](#versioning) section below for more info.
```

Below it says "Our aim is to get to a 1.0.0 final by the end of August 2023.", so maybe it's better to give a more definitive timeline to avoid people saying they wanted to provide feedback the first week of September? ;-)

m-mohr reviewed (commented):
All the 1.0.0-dev occurances get updated automatically, right?

m-mohr commented on README.md:
```diff
@@ -2,19 +2,19 @@
 - [**Specification**](format-specs/geoparquet.md)
 - [JSON Schema](format-specs/schema.json)
 - [Examples](examples/)
-- [Validator](validator/)
```
The link is getting removed but the folder is still in the repo. That feels kinda weird.

cholmes replied:
Yeah, we opened https://github.com/geoparquet

It's on my list to move it out and over, but didn't want to block rc.1 on that.
